spec/unit/remotes_spec.rb

```diff
@@ -1,39 +1,159 @@
 require '3scale_toolbox'
 
 RSpec.describe ThreeScaleToolbox::Remotes do
-  let(:test_class) do
-    Class.new { include ThreeScaleToolbox::Remotes }
+  include_context :resources
+  include_context :temp_dir
+
+  RSpec.shared_examples 'raises invalid url error' do
+    it 'raises error' do
+      expect { subject.add_uri('some_name', origin) }.to raise_error(ThreeScaleToolbox::InvalidUrlError,
+                                                                     /invalid url/)
+    end
   end
 
-  subject { test_class.new }
+  let(:yaml_content) do
+    <<~HEREDOC
+      ---
+        :remotes:
+          remote_1:
+            :endpoint: https://1.example.com
+            :auth_key: '123456789'
+    HEREDOC
+  end
+  let(:config_file) do
+    File.join(tmp_dir, '.3scalerc').tap do |config_file|
+      File.open config_file, 'w' do |file|
+        file.write(yaml_content)
+      end
+    end
+  end
+  let(:config) { ThreeScaleToolbox::Configuration.new(config_file) }
+  let(:endpoint) { 'https://1.example.com' }
+  let(:authkey) { '123456789' }
```

Comment:
There is no concept of `authkey` in 3scale.

* access_token
* provider_key

Or more general terms for Basic auth:

* user
* password
* userinfo

https://tools.ietf.org/html/rfc3986#section-3.2.1

Reply:
I got the name from https://github.com/3scale/3scale_toolbox/pull/53#discussion_r230045161

That is a key given to 3scale client. According to 3scale client constructor:
https://github.com/3scale/3scale-api-ruby/blob/585c09f69ab57e62a48d17d05e9132c275719f6e/lib/3scale/api.rb#L8

Some `provider_key` has to be given. Since `access_token`s are also valid, IMO it seems `token` or `auth_key` to be more appropriate. In the request whether it is user information or auth header or some other kind of token is implementation detail. 

What do you propose? `user` or `password` does not seem to fit very well. `userinfo`??

Reply:
Sorry, do not see the difference between `authkey` and `authentication`. Moreover, IMO `authkey` is more meaninful. `authentication` is more about a method or process for identifying a user, while `authkey` is about id item of that process. 

About changing the format remote credentials are stored, IMO this is out of scope of this PR. Happy to discuss about formats. Having specific field for the token or having as part of the userinfo in the url.

Reply:
The problem with `authkey` is that is is made up word and similar to `app_key` that is already used by 3scale.

Reply:
changed to `authentication` :+1: